src/pydicom/dataelem.py

```diff
@@ -522,13 +522,13 @@ def _convert_value(self, val: Any) -> Any:
             return Sequence(val)
 
         # if the value is a list, convert each element
-        try:
-            val.append
-        except AttributeError:  # not a list
+        if not hasattr(val, "append"):
```

Comment:
A failing exception is actually a little bit expensive and VM = 1 is the most common case, so changed to `hasattr`